Application/Commands/Enquiry/Build/AddEnquiryCommand.cs

```diff
@@ -310,12 +378,16 @@ private async Task HandleDbUpdateException(DbUpdateException ex, Domain.Enquiry
                 _logger.LogError(
                     ex,
                     "Violation on unique constraint. Support Reference Number: {referenceNumber}.  Next retry attempt number: {retryAttempt}",
-                    enquiry.SupportReferenceNumber, retryAttempt);
+                    _enquiryReferenceNumber, retryAttempt);
 
-                enquiry.SupportReferenceNumber = _generateReferenceNumber.GenerateReferenceNumber();
+                _unitOfWork.EnquiryRepository.RollbackChanges();
```

Comment:
The problem is the way that EF maintains the Entity changed state, throwing the error does not roll back the state.  We're rolling back the state of the data so we are back to initial state and can then get a new ref number and call `GetEnquiry` again to populate the data from scratch.  It doesn't work without this and I believe it would need significant changes to the whole class to restructure it to work to re-populate the data in a different way.